src/context/AppProvider/AppContext.tsx

```diff
@@ -294,31 +292,37 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
    * fetch forex spot and history for user's selected currency
    */
   const currency = useAppSelector(state => selectSelectedCurrency(state))
-  const timeframe = DEFAULT_HISTORY_TIMEFRAME
-  const priceHistoryArgs = useMemo(() => ({ symbol: currency, timeframe }), [currency, timeframe])
-  const { error: fiatPriceHistoryError } = useFindPriceHistoryByFiatSymbolQuery(priceHistoryArgs)
-  const { error: forexRateError } = useFindByFiatSymbolQuery(priceHistoryArgs)
 
   useEffect(() => {
-    /**
-     * crypto market data is denominated in USD and is the "safe" condition we can
-     * recover from failures on
-     */
+    // we already know 1usd costs 1usd
     if (currency === 'USD') return
-    if (fiatPriceHistoryError || forexRateError) {
-      toast({
-        position: 'top-right',
-        title: translate('multiCurrency.toast.title', { symbol: currency }),
-        description: translate('multiCurrency.toast.description'),
-        status: 'error',
-        duration: null, // don't auto-dismiss
-        isClosable: true,
-      })
-      dispatch(preferences.actions.setSelectedCurrency({ currency: 'USD' }))
-    }
+
+    void (async () => {
+      const timeframe = DEFAULT_HISTORY_TIMEFRAME
+      const priceHistoryArgs = { symbol: currency, timeframe }
+      const { error: fiatPriceHistoryError } = await dispatch(
+        marketApi.endpoints.findPriceHistoryByFiatSymbol.initiate(priceHistoryArgs),
+      )
+      const { error: forexRateError } = await dispatch(
+        marketApi.endpoints.findByFiatSymbol.initiate(priceHistoryArgs),
+      )
+
+      if (fiatPriceHistoryError || forexRateError) {
+        toast({
+          position: 'top-right',
+          title: translate('multiCurrency.toast.title', { symbol: currency }),
+          description: translate('multiCurrency.toast.description'),
+          status: 'error',
+          duration: null, // don't auto-dismiss
+          isClosable: true,
+        })
+        dispatch(preferences.actions.setSelectedCurrency({ currency: 'USD' }))
+      }
+    })()
+
     // setting symbol causes infinite render
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [dispatch, fiatPriceHistoryError, forexRateError, toast])
```

Comment:
Since these two were effectively depending themselves on `currency` and `timeframe`, we may want to bring `currency` back here explicitly - switching to errorred other currencies should toast, but doesn't and keeps the same USD (non-prorated) value

<img width="751" alt="image" src="https://github.com/shapeshift/web/assets/17035424/21aa5c0e-08e8-4bd9-8a22-8c6572490f58">
<img width="758" alt="image" src="https://github.com/shapeshift/web/assets/17035424/9cb61b84-1593-4a8e-83ad-423283623e5b">

Reply:
nice find. confirmed this fixes the above issue and that infinite render is not triggered as when currency is set to USD the function exits early

Reply:
✅ https://github.com/shapeshift/web/pull/5346/commits/8c39836a466abbafdb577a9f30d2b4b3a63c83a0